### PR TITLE
Sign-out feature

### PR DESCRIPTION
**Pull request by PH-14**
This PR adds a sign-out button on the profile page in order to sign out the user and redirect him/her towards the authentication activity.

### CODE REVIEW

File: app/src/main/java/com/github/sdp/mediato/ui/MyProfileFragment.java [hunk 1]
```diff
@@ -32,140 +38,146 @@
...
+        photoPicker = setupPhotoPicker();
+        collectionlistAdapter = setupCollections(collectionListRecyclerView);
+        setupAddCollectionsButton(addCollectionButton);
+
+        // Observe the view model's live data to update UI components
+        observeCollections(collectionlistAdapter);
+
+        // Add on click listener to sign out button
+        Button signOutButton = view.findViewById(R.id.signout_button);
+        signOutButton.setOnClickListener(v -> {((MainActivity) getActivity()).signOutUser();});
```
Comment (simone-kalbermatter) [inline app/src/main/java/com/github/sdp/mediato/ui/MyProfileFragment.java:77]: I think the curly braces here are not needed

File: app/src/main/res/layout/layout_profile_header.xml [hunk 2]
```diff
@@ -10,6 +11,22 @@
   app:layout_constraintTop_toTopOf="parent"
   app:layout_constraintVertical_bias="0.0">
 
+  <Button
```
Comment (simone-kalbermatter) [inline app/src/main/res/layout/layout_profile_header.xml:14]: Could you change the color and size of the button to make it a bit less eye catching for a sign out button 😄 And maybe change the text to _sign out_ or just a sign out icon to make it a bit more clear what the button does.

File: app/src/main/res/layout/layout_profile_header.xml [hunk 3]
```diff
@@ -10,6 +11,22 @@
   app:layout_constraintTop_toTopOf="parent"
   app:layout_constraintVertical_bias="0.0">
 
+  <Button
+      android:id="@+id/signout_button"
+      android:layout_width="90dp"
+      android:layout_height="50dp"
```
Comment (simone-kalbermatter) [inline app/src/main/res/layout/layout_profile_header.xml:17]: It would be better to use `"wrap_content"` instead of hardcoding the button size.

**Review by simone-kalbermatter — CHANGES_REQUESTED** [review]
LGTM, clean work and good tests. Just a few comments. And I think there is some formatting issue because some full files are shown as diff.

File: app/src/main/java/com/github/sdp/mediato/ui/MyProfileFragment.java [hunk 4]
```diff
@@ -32,140 +38,146 @@
...
+                             Bundle savedInstanceState) {
+        viewModel = ((MainActivity) getActivity()).getCurrentUserViewModel();
+
+        // Initializes the profile header, based on USERNAME
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+
+        // Get all UI components
+        editButton = view.findViewById(R.id.edit_button);
+        addCollectionButton = view.findViewById(R.id.add_collection_button);
+        collectionListRecyclerView = view.findViewById(R.id.collection_list_recycler_view);
```
Comment (MeKHell) [inline app/src/main/java/com/github/sdp/mediato/ui/MyProfileFragment.java:65]: You might want to look into binders to remove the need of these lines.

**Review by MeKHell — COMMENTED** [review]
I do not know what soft you use to commit your change, but it is pretty hard to know what you changed and what you did not.

**Review by simone-kalbermatter — APPROVED** [review]
LGTM!

**Pull request merged**